addons/storyshots/src/index.js

```diff
@@ -21,6 +27,16 @@ const hasDependency = name =>
   (pkg.devDependencies && pkg.devDependencies[name]) ||
   (pkg.dependencies && pkg.dependencies[name]);
 
+function patchStoriesOfFunction(storybookImpl) {
```

Comment:
Could we just save `filename` alongside the story in core rather than doing this hack?